pytensor/tensor/subtensor.py

```diff
@@ -197,18 +230,37 @@ def get_canonical_form_slice(
     in which 0 <= start <= stop <= length and step > 0, and a flag which says
     if the resulting set of numbers needs to be reversed or not.
 
+    Returns
+    -------
+    slc
+        Slice or scalar variable.
+    direction
+        Direction to iterate the resulting elements in. (-1 or 1).
     """
     from pytensor.tensor import ge, lt, sign, switch
 
     if not isinstance(theslice, slice):
-        try:
-            value = as_index_literal(theslice)
-        except NotScalarConstantError:
-            value = theslice
-
-        value = switch(lt(value, 0), (value + length), value)
+        # Try to extract a scalar slice and return it already.
+        sslice: int | np.integer | ScalarVariable | None = None
+        if isinstance(theslice, ScalarVariable):
+            sslice = theslice
+        else:
+            try:
+                vlit = as_index_literal(theslice)
+                if vlit is None:
+                    return slice(0, length, 1), 1
```

Comment:
Are you sure? Isn't `None` a `newaxis`, which shouldn't correspond to any slice?

Reply:
Also unclear whether this function is expected to handle None. Like SliceVariable, those only show up in the index for AdvancedSubtensor Ops, for basic Subtensor Ops, newaxis are eagerly converted to `expand_dims`, and never show as inputs of the Op